web-marketplace/src/components/molecules/CreditsAmount/CurrencyInput.tsx

```diff
@@ -39,22 +39,18 @@ export const CurrencyInput = ({
   } = useFormContext<ChooseCreditsFormSchemaType>();
   const { _ } = useLingui();
 
-  const { onChange } = register(CURRENCY_AMOUNT);
-
   const currency = useWatch({
     control,
     name: CURRENCY,
   });
 
   const handleOnChange = useCallback(
     (event: ChangeEvent<HTMLInputElement>) => {
-      // Remove zeros in non decimal values and update the value
-      const value = event.target.value;
-      if (!value.includes('.')) setValue(CURRENCY_AMOUNT, Number(value));
```

Comment:
removing this introduces another bug, you can't type a number with 2 decimals if the first decimal is 0, eg 21,05

Reply:
Reverting doesn't seem to work either. There seems to be an error that may be related to stripe when introducing numbers 2.01. Any idea what the thrown error is about @blushi . Also the user gets stuck in the error screen because the amount introduced is in the Local Storage. 

https://github.com/user-attachments/assets/5138d4d8-3f61-43f4-9af6-71c40968c975

Reply:
I think that should be a separated issue btw. If we are happy with the fix for the current issue we may want to merge this so other features don't have to deal with not being able to use the currency input.

Reply:
ok, thanks -  I have added a possible solution to this too